### PR TITLE
[MPDX-7894] Save all of the user's past account list ids in the DataDog session

### PR DESCRIPTION
**Pull request by canac**
## Description

* Since `useDataDog.ts` doesn't export a hook, rename it to `dataDog.ts` and move it from `hooks` to `lib`.
* Remember the user's previous account list ids and add all of them to the DataDog session.

This PR is not immediately related to https://jira.cru.org/browse/MPDX-7894. However it lays the foundation for adding the user's language in a follow-up PR, which is relevant to that ticket.

## Checklist:

- [x] I have given my PR a title with the format "MPDX-(JIRA#) (summary sentence max 80 chars)"
- [x] I have applied the appropriate labels. (_Add the label "On Staging" to get the branch automatically merged into staging._)
- [x] I have requested a review from another person on the project

### CODE REVIEW

File: src/lib/dataDog.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  interface Window {
+    DD_RUM: {
+      setUser: (user: Record<string, unknown>) => void;
+      clearUser: () => void;
+    };
+  }
+}
+
+export const isDataDogConfigured = (): boolean => {
+  if (typeof window === 'undefined') return false;
```
Comment (wrandall22) [inline src/lib/dataDog.ts:11]: I think we should adopt the `curly: ['error', 'all']` lint rule. I'm a huge fan of always having curly braces around if blocks, even if they are one liners. Feel free to discuss. cc @dr-bizz

Comment (canac) [reply src/lib/dataDog.ts:11]: I'm used to the curly braces being everywhere as well, and it always throws me off a little bit when they're not there.

Comment (canac) [reply src/lib/dataDog.ts:11]: #945

File: src/components/DataDog/DataDog.tsx [hunk 2]
```diff
@@ -11,11 +11,11 @@ const DataDog: React.FC = () => {
     ? Array.isArray(query.accountListId)
       ? query.accountListId[0]
       : query.accountListId
-    : '';
+    : null;
 
   const user = session?.user;
   useEffect(() => {
-    if (user) {
+    if (user && accountListId) {
```
Comment (dr-bizz) [inline src/components/DataDog/DataDog.tsx:18]: What if the user is on the accountListId page when they have mulitple accounts. Currently DataDog won't be saving their session

File: src/lib/dataDog.ts (hunk 1 above)
Comment (dr-bizz) [reply src/lib/dataDog.ts:11]: I'm good with adding it

File: src/lib/dataDog.test.ts [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    });
+  });
+
+  describe('when DataDog is configured', () => {
+    beforeEach(() => {
+      process.env.DATADOG_CONFIGURED = 'true';
+    });
+
+    //#region Default Tests
+    it('isDataDogConfigured should return false', () => {
```
Comment (dr-bizz) [inline src/lib/dataDog.test.ts:41]:
```suggestion
    it('isDataDogConfigured should return true', () => {
```

**Review by dr-bizz — COMMENTED** [review]
I had a few questions, but overall it looks great. Thank you for working on this.

File: src/lib/dataDog.test.ts (hunk 3 above)
Comment (canac) [reply src/lib/dataDog.test.ts:41]: Good catch, thanks! I must have accidentally changed that.

File: src/components/DataDog/DataDog.tsx (hunk 2 above)
Comment (canac) [reply src/components/DataDog/DataDog.tsx:18]: You're right, I need to remove that `&& accountListId` condition. Thanks for catching that.

**Review by dr-bizz — APPROVED** [review]
The code looks great! Thank you for implementing this

**Pull request merged**